### DISCUSSION TITLE
Authorization services possible separation of individuals stores

### DISCUSSION THREAD

**Discussion by vramik** [keycloak/keycloak#11662] [Archive]
I'd like to get confirmation there is no plan to support different stores for each authz object (`Permission`, `Scope`, `ResourceServer`, ...) in future. AFAIK there is not, but I want to be sure.

When implementing jpa-map-storage for authz I've encountered following code in  `ModelEntityUtil.MODEL_TO_NAME`

```
// authz
MODEL_TO_NAME.put(PermissionTicket.class, "authz-permission-tickets");
MODEL_TO_NAME.put(Policy.class, "authz-policies");
MODEL_TO_NAME.put(ResourceServer.class, "authz-resource-servers");
MODEL_TO_NAME.put(Resource.class, "authz-resources");
MODEL_TO_NAME.put(org.keycloak.authorization.model.Scope.class, "authz-scopes");
```
I'd like to have all changelogs for all of them into single one so I was wonding if we should/could change `ModelEntityUtil.MODEL_TO_NAME` following way:

```
// authz
MODEL_TO_NAME.put(PermissionTicket.class, "authz");
MODEL_TO_NAME.put(Policy.class, "authz");
MODEL_TO_NAME.put(ResourceServer.class, "authz");
MODEL_TO_NAME.put(Resource.class, "authz");
MODEL_TO_NAME.put(org.keycloak.authorization.model.Scope.class, "authz");
```

Would it break something? I can workaround it in the jpa implementation if there is an issue with it.

**Comment by mhajas** [comment]
Thanks for bringing this up @vramik. From HotRod point of view, there is no problem with it. I actually did [the same thing](https://github.com/keycloak/keycloak/pull/10066/files#diff-1453f65178bef4ac0f4b5569cf7c894c183580f5217314ff55f334fec50dce64R121-R169) when started work on HotRod authz implementation. 

The problem may be tree store configuration as if I understand correctly, we are now able to configure different tree/node configurations for each authz sub-area. But if we decide we want only one configuration for the whole authz, this change could be ok.

**Comment by vramik** [comment]
There are stores stored in CHM based on the names: https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/storage/chm/ConcurrentHashMapStorageProviderFactory.java#L97 

To make it work we'd need to change CHM implementation to something like `ConcurrentHashMap<Class<?>, ConcurrentHashMapStorage<?,?,?>>` and it imples additional changes like `keyKonvertor` etc.

Therefore we've decided we'll park it for now and adjust it only in jpa implementation, similarly as it is in HotRod currently.
